utils/src/com/cloud/utils/Profiler.java

```diff
@@ -23,24 +23,20 @@
     private Long startTickInMs;
     private Long stopTickInMs;
 
-    public Profiler() {
-        startTickInMs = null;
-        stopTickInMs = null;
-    }
-
     public long start() {
-        startTickInMs = System.currentTimeMillis();
+        startTickInMs = System.nanoTime();
         return startTickInMs.longValue();
     }
 
     public long stop() {
-        stopTickInMs = System.currentTimeMillis();
+        stopTickInMs = System.nanoTime();
         return stopTickInMs.longValue();
     }
 
     public long getDuration() {
-        if (startTickInMs != null && stopTickInMs != null)
+        if (startTickInMs != null && stopTickInMs != null) {
             return stopTickInMs.longValue() - startTickInMs.longValue();
```

Comment:
Yes, and so does the semantics of the return of starts() stop() and getDuration().
I checked; the first two are never cought in a variable but the last one is used in output and to check against settings. There the semantics really changes. It now returns higher numbers then axpected.